packages/dev/serializers/src/glTF/2.0/glTFAnimation.ts

```diff
@@ -943,37 +965,37 @@ export class _GLTFAnimation {
     }
 
     private static _GetBasePositionRotationOrScale(
-        babylonTransformNode: TransformNode,
+        babylonTransformNode: TransformableType,
         animationChannelTargetPath: AnimationChannelTargetPath,
         convertToRightHandedSystem: boolean,
         useQuaternion: boolean
     ) {
         let basePositionRotationOrScale: number[];
         if (animationChannelTargetPath === AnimationChannelTargetPath.ROTATION) {
             if (useQuaternion) {
-                if (babylonTransformNode.rotationQuaternion) {
-                    basePositionRotationOrScale = babylonTransformNode.rotationQuaternion.asArray();
-                    if (convertToRightHandedSystem) {
-                        _GLTFUtilities._GetRightHandedQuaternionArrayFromRef(basePositionRotationOrScale);
-                        if (!babylonTransformNode.parent) {
-                            basePositionRotationOrScale = Quaternion.FromArray([0, 1, 0, 0]).multiply(Quaternion.FromArray(basePositionRotationOrScale)).asArray();
-                        }
+                const q: Quaternion = (<any>babylonTransformNode).rotationQuaternion;

```

Comment:
I guess the union type didn't actually help right?

It's not perfect, but I would probably do this to avoid using `any`. One of the problems with using `any` is that you can very easily spell something wrong. Same for the other properties below.

```suggestion
                const q = (babylonTransformNode as TransformNode).rotationQuaternion;
```

Reply:
Yes you're right, the problem came from we use other object which are not `TransformNode`, `Camera` and `Light` for instance.
However these objects may have, down the hierarchy, similar properties such `rotation` or `rotationQuaternion` or `position` and Ideally they should share a common interface which encompass these properties.
Type assertion will not change the game, but i'm agree that we may better put something else that `any`. Maybe introduce an interface instead of union type ?